openssl/src/bn.rs

```diff
@@ -1401,4 +1421,37 @@ mod tests {
+        let element_size = 1500;
+        let mut accumulator = BigNum::from_dec_str("1915182769368569486963976668657418846122016471585262452138689064979880063217142171230525667406114297912997697884669760718181148903891406258652154564474937314809392827351229074850339882653085906709452763512017926382658040877860425398565313118206001571983057266861433787870120516963262872556160137632986790799637800187327525373894833699998137974425675888510468704041321563743266953919273330110447213516904474856325884904918747921700098338253140217925656078220294833113261362981479227102302139442862538529417112227112104830922667359007109709204936925922862592053626140248383988977967399826184116697852122995719423774656923017777301713520425580745782097460248899329835740441429284020663186635354571333690692107185946297929772630086928323118606594588790915139339431940489511850279806818826143040378319619062305629273497273725721368157955512179352674013822087432127910233025333571203452180390613181846035364105352384024395662959142236311712295552933323767329374261596612060957502167831080774021620823718240397211269873903737841346862215246685242302847894894907938934985950908016181737929997451061016202494801606490097096889822976223609532834237496884730642312786447889763148272139117981855545441295507862786712405210727374768044118436617961").unwrap();
+
+        // Number of primes to exponenetiate
+        let num_exponentiations = 100;
+
+        let mut primes = Vec::new();
+        let start_since_the_epoch = SystemTime::now().duration_since(UNIX_EPOCH)
+            .expect("Time went backwards");
+        for _ in 0..num_exponentiations {
+            let mut p = BigNum::new().unwrap();
+            p.generate_prime(element_size, false, None, None).unwrap();
+            primes.push(p);
+        }
+        let end_since_the_epoch = SystemTime::now().duration_since(UNIX_EPOCH)
```

Comment:
`SystemTime` shouldn't be used for timing because of the time shift issue. `Instant` doesn't have those problems.

Reply:
You're right, also `duration_since(UNIX_EPOCH)` seems wasteful. I threw this test up in a rush.